ui/app/src/components/workspaces/WorkspaceUsersAssignNew.tsx

```diff
@@ -0,0 +1,142 @@
+
+  const navigate = useNavigate();
+  const apiCall = useAuthApiCall();
+
+  const [roleOptions, setRoleOptions] = useState<IDropdownOption[]>([]);
+
+  const [selectedUser, setSelectedUser] = useState<string | null>(null);
+  const [selectedRole, setSelectedRole] = useState<string | null>(null);
+
+  const [assigning, setAssigning] = useState(false);
+  const [hasAssignmentError, setHasAssignmentError] = useState(false);
+  const [assignmentError, setAssignmentError] = useState({} as APIError);
+
+  const onUserChange = (event: any) => {
+    setSelectedUser(event ? event.target.value : null);
+  };
+
+  const onRoleChange = (event: any, option: any) => {
+    setSelectedRole(option ? option.key : null);
+  };
+
+  const dismissPanel = useCallback(() => navigate('../'), [navigate]);
+
+  const getWorkspaceRoles = useCallback(async () => {
+    try {
+      const scopeId = "";
+      const response = await apiCall(`${ApiEndpoint.Workspaces}/${workspace.id}/${ApiEndpoint.Roles}`, HttpMethod.Get, scopeId);
+
+      const options: IDropdownOption[] = response.roles.map((workspaceRole: WorkspaceRole) => ({
+        key: workspaceRole.value,
+        text: workspaceRole.displayName
+      }));
+
+      setRoleOptions(options);
+    }
+    catch (err: any) {
+      err.userMessage = 'Error retrieving assignable users';
+    }
+
+  }, [apiCall, workspace.id]);
+
+  useEffect(() => {
+    getWorkspaceRoles();
+  }, [getWorkspaceRoles]);
+
+  const assign = useCallback(async () => {
+    setAssigning(true);
+
+    const encodedUser=selectedUser?.replaceAll('#', '%23');
+
+    const scopeId = "";
+    try {
+      const response = await apiCall(`${ApiEndpoint.Workspaces}/${workspace.id}/${ApiEndpoint.Users}/assign?user_email=${encodedUser}&role_name=${selectedRole}`, HttpMethod.Post, scopeId);
+      props.onAssignUser(response);
+    }
+    catch (err: any) {
+      err.userMessage = 'Error assigning workspace user';
+      setHasAssignmentError(true);
+      setAssignmentError(err);
+    }
+    setAssigning(false);
+
+  }, [selectedUser, apiCall, workspace.id, selectedRole, props]);
+
+  const renderFooter = useCallback(() => {
+    let footer = <></>
+    footer = <>
+      <div style={{ textAlign: 'end' }}>
+        <PrimaryButton onClick={() => assign()} disabled={assigning || (!selectedUser || !selectedRole)}>Assign</PrimaryButton>
+      </div>
+    </>
+    return footer;
+  }, [selectedUser, selectedRole, assign, assigning]);
+
+  return (
+    <Panel
+      headerText="Assign user to a role"
+      isOpen={true}
+      isLightDismiss={true}
+      onDismiss={dismissPanel}
+      onRenderFooterContent={renderFooter}
+      isFooterAtBottom={true}
+      closeButtonAriaLabel="Close"
+      type={PanelType.custom}
+      customWidth="450px"
+    >
+      <Stack tokens={{ childrenGap: 20 }} styles={{ root: { paddingTop: 20 } }}>
+        <Stack tokens={{ childrenGap: 10 }} verticalAlign="center">
+          <Label>User</Label>
+          <TextField
```

Comment:
Is it possible to have a picker here of some sort, I tried with an email in the tenant and got a 500 back. Didn't dig any deeper.

Reply:
Originally I had a picker here but it didn't look great with large numbers in the tenant.  Ideally what it needs is some sort of search functionality. For external users to a tenant it needs to be added as the user principal name rather than the email address so it is a bit clunky.  (e.g. myemail_microsoft.com#EXT#@MyTenant.onmicrosoft.com).

I can readd the name picker as a short term measure and investigate the search functionality as a longer term solution (possibly a new issue)?

Reply:
Not sure what the picker looks like... got a snip?

Yes, agree, search would work better.

Reply:
![image](https://github.com/user-attachments/assets/ee719416-6a6d-42ed-9d97-fd9f0ced9b24)

This was the initial attempt, for each user we show the Display Name and the User Principal Name (email address for non-guest users)

Reply:
It works fine while the GAL is small.

Reply:
Nice, does it filter as you type or not? 

I think add it, and we can maybe modify as we go on. Is it an existing control/component - https://developer.microsoft.com/en-us/fluentui#/controls/web/peoplepicker ?

Reply:
No it is a combobox with an API, will add it now and take a look at that link to see if I can get it in easily?

Reply:
This is now replaced with the people picker